File: Pilha.java
import java.util.ArrayList;

public class Pilha {
    private ArrayList<String> elementos;

    // Construtor
    public Pilha() {
        elementos = new ArrayList<>();
    }

    // Inserir elemento no topo da pilha
    public void push(String item) {
        elementos.add(item);
    }

    // Remover elemento do topo
    public String pop() {
        if (!isEmpty()) {
            return elementos.remove(elementos.size() - 1);
        }
        return "Pilha vazia!";
    }

    // Consultar o topo da pilha
    public String peek() {
        if (!isEmpty()) {
            return elementos.get(elementos.size() - 1);
        }
        return "Pilha vazia!";
    }

    // Verificar se a pilha está vazia
    public boolean isEmpty() {
        return elementos.isEmpty();
    }

    // Mostrar elementos da pilha
    public void mostrarPilha() {
        System.out.println("Pilha: " + elementos);
    }

    public static void main(String[] args) {
        Pilha pl = new Pilha();

        pl.push("António");
        pl.push("Armando");
        pl.push("Janeta");
        pl.push("Azaiel");

        pl.mostrarPilha();

        System.out.println("Topo: " + pl.peek());
        System.out.println("Removido: " + pl.pop());
        System.out.println("Novo topo: " + pl.peek());
        System.out.println("Está vazia? " + pl.isEmpty());

        pl.mostrarPilha();
    }
}

File: pilha.py
class Pilha:
    def __init__(self):
        self.elementos = []

    # Inserir no topo da pilha
    def push(self, item):
        self.elementos.append(item)

    # Remover do topo
    def pop(self):
        if not self.is_empty():
            return self.elementos.pop()
        return "Pilha vazia!"

    # Consultar topo
    def peek(self):
        if not self.is_empty():
            return self.elementos[-1]
        return "Pilha vazia!"

    # Verificar se está vazia
    def is_empty(self):
        return len(self.elementos) == 0

    # Mostrar pilha
    def mostrar_pilha(self):
        print("Pilha:", self.elementos)


if __name__ == "__main__":
    pilha = Pilha()

    pilha.push("António")
    pilha.push("Armando")
    pilha.push("Janeta")
    pilha.push("Azaiel")

    pilha.mostrar_pilha()

    print("Topo:", pilha.peek())
    print("Removido:", pilha.pop())
    print("Novo topo:", pilha.peek())
    print("Está vazia?", pilha.is_empty())

    pilha.mostrar_pilha()